libs/watt/package/yearmonth-field/watt-yearmonth-field.component.ts

```diff
@@ -0,0 +1,178 @@
+        position-try-fallbacks: flip-block;
+        width: 296px;
+        height: 354px;
+        inset: unset;
+        margin: unset;
+        border: 0;
+      }
+    `,
+  ],
+  template: `
+    <watt-field [label]="label()" [control]="control" [anchorName]="anchorName">
+      <input
+        #field
+        readonly
+        [formControl]="control"
+        (focus)="handleFocus(picker)"
+        (blur)="handleBlur(picker, $event)"
+      />
+      <watt-button icon="date" variant="icon" (click)="field.focus()" />
+      <div
+        #picker
+        class="watt-elevation watt-yearmonth-field-picker"
+        popover="manual"
+        tabindex="0"
+        [style.position-anchor]="anchorName"
+      >
+        @if (isOpen()) {
+          <mat-calendar
+            startView="multi-year"
+            [startAt]="selected()"
+            [selected]="selected()"
+            [minDate]="min()"
+            [maxDate]="max()"
+            (monthSelected)="handleSelectedChange(field, $event)"
+          />
+        }
+      </div>
+      <ng-content />
+      <ng-content select="watt-field-error" ngProjectAs="watt-field-error" />
+      <ng-content select="watt-field-hint" ngProjectAs="watt-field-hint" />
+    </watt-field>
+  `,
+})
+export class WattYearMonthField implements ControlValueAccessor {
+  // Popovers exists on an entirely different layer, meaning that for anchor positioning they
+  // look at the entire tree for the anchor name. This gives each field a unique anchor name.
+  private static instance = 0;
+  private instance = WattYearMonthField.instance++;
+  protected anchorName = `--watt-yearmonth-field-popover-anchor-${this.instance}`;
+
+  // The format of the inner FormControl is different from that of the outer FormControl
+  protected control = new FormControl('', { nonNullable: true });
+
+  // `registerOnChange` may subscribe to this component after it has been destroyed, thus
+  // triggering an NG0911 from the `takeUntilDestroyed` operator. By sharing the observable,
+  // the observable will already be closed and `subscribe` becomes a proper noop.
+  private yearMonthChanges = this.control.valueChanges.pipe(map(YearMonth.fromView));
+  private valueChanges = this.yearMonthChanges.pipe(
+    map((yearMonth) => yearMonth.toModel()),
+    takeUntilDestroyed(),
+    share()
+  );
+
+  private yearMonth = toSignal(this.yearMonthChanges);
+  protected selected = computed(() => this.yearMonth()?.toDate());
+
+  // This is used to reset the MatCalendar component by destroying and then recreating it
+  // whenever the picker is opened. There is no methods to do it programatically.
+  protected isOpen = signal(false);
+
+  /** Set the label text for `watt-field`. */
+  label = input('');
+
+  /** The minimum selectable date. */
+  min = input<Date>();
+
+  /** The maximum selectable date. */
+  max = input<Date>();
+
+  /** Emits when the selected month has changed. */
+  monthChange = outputFromObservable(this.valueChanges);
```

Comment:
Question: What use case would you need an event on month changed? Would you not use the formcontrols valuechanged event?

Reply:
Yes usually, this just makes it possible to use the component without a `FormControl` for simpler use cases. Also handy if you need to trigger something directly from the template in response to the change event. It is just one line of code that adds some convenience and I see the same pattern in some of the Angular Material components, which is why I included it.